model.py

```diff
@@ -6,147 +6,149 @@
 from keras.preprocessing.text import Tokenizer
 from tensorflow import keras
 
+from parseargs import ParseArgs
 from utils import save_embeddings
 from visualization import plot_loss, plot_accuracy
 
-vocab_size = 10000
 
+class Model:
```

Comment:
I don't really see why this is a class. There's no data associated with each Model instance - it's just a collection of functions.

Reply:
As is, the interface is a bit confusing.

I'd suggest bundling the keras model and corpus vocabulary together - it's not appropriate to use a model with a different corpus than it was trained on anyway (as I understand), so don't let them get separated.

You could do that by expanding the model class to contain all the relevant data.

Reply:
Anyway, then the interface might look like this:

```python
model = build_model(lots of parameters)
save_model(model, 'filename')

# later
model = load_model('filename')

# In use:
model.predict('sentence')
```

build/save/load_model could be static methods of the model class or just module functions.

Reply:
Here what you're saying, I did this so the interface is enforced whenever another class wants to interact with the anything to do with the model. Shall rethink this in the future